libs/sync/src/lib/upload/tabs/edit-settings.component.ts

```diff
@@ -41,15 +41,28 @@ export class EditSyncSettingsComponent implements OnInit {
   }
 
   save() {
-    if (this.sync) {
+    const sync = this.sync;
+    if (sync) {
       this.syncService
-        .updateSync(this.sync.id, {
-          title: this.form?.value['title'] || '',
-          connector: {
-            ...this.sync.connector,
-            parameters: { ...this.sync.connector.parameters, ...(this.form?.value['fields'] || {}) },
-          },
-        })
+        .getConnector(sync.connector.name, '')
+        .pipe(
+          switchMap((sourceConnector) => {
+            const payload: Partial<ISyncEntity> = {
+              title: this.form?.value['title'] || '',
+              connector: {
+                ...sync.connector,
+                parameters: { ...sync.connector.parameters, ...(this.form?.value['fields'] || {}) },
+              },
+            };
+            if (!sourceConnector.allowToSelectFolders) {
+              if (typeof sourceConnector.handleParameters === 'function') {
+                sourceConnector.handleParameters(sync.connector.parameters);
+              }
+              payload.foldersToSync = sourceConnector.getStaticFolders();
```

Comment:
It seems that `sourceConnector.getStaticFolders()` still return the old folder.

Reply:
fixed
thanks!